src/services/ShSimulatorService.tsx

```diff
@@ -447,12 +540,70 @@ const ShSimulation = ({ children }: GenericContextProviderProps) => {
 		[getJobStatus]
 	);
 
+	const findFirstEstimatorName = useCallback(
```

Comment:
this method should be well documented with **comments** so the **reader of the code** knows what happens when we load results of the simulation which has already completed execution and what happens if we load some type of JSON like project with results (like example)

Reply:
this comment says nothing on the major difference between "from user files" and "from editor" simulations. As a reader of that code and an user of the platform I would expect that to be explained here.

It should be explicitely mention that you expect to have `inputJson?.scoringManager.outputs[0].name` only for "from editor" simulation and that this field contains Outputs/Estimator names specified by user in the Editor.

Reply:
You should also point user to the code where `inputJson?.scoringManager.outputs[N].name` is being read to get estimator name for estimator number N